labkey-client-api/src/org/labkey/remoteapi/Command.java

```diff
@@ -491,11 +485,25 @@ private URI createURI(Connection connection, String folderPath) throws URISyntax
             path.append(folderPathNormalized);
         }
 
+        if (path.charAt(path.length() - 1) != '/')
+            path.append('/');
+
+        //add the controller name
+        String controller = getControllerName();
+        //for now, strip leading /. TODO: For next major release, throw
+        if (controller.charAt(0) == '/')
+            controller = controller.substring(1);
+        //for now, strip trailing /. TODO: For next major release, throw
+        if (controller.charAt(controller.length() - 1) == '/')
```

Comment:
Maybe consider using StringUtils.removeStart(), StringUtils.appendIfMissing() etc to make more concise.  (Or just clone org.labkey.api.util.Path which is slightly overkill, but it's not a big file)

Reply:
I'd love to, but `StringUtils` isn't a dependency :(